src/csrfHeader.js

```diff
@@ -4,15 +4,16 @@
  * @providesModule csrfHeader
  * @flow
  */
+import type { CustomHeaders } from './types'
 
-function getToken(): ?string {
+export function getToken(): ?string {
   const meta = document.querySelector(`meta[name="csrf-token"]`)
   return meta && meta.getAttribute('content')
 }
 
-export default function csrfHeader(): { 'X-CSRF-Token': string } {
+export default function csrfHeader(headers?: CustomHeaders): CustomHeaders {
```

Comment:
I think this is a little more complex and entangled than is necessary. What do you think of this as a simple getter: what csrf header object should we add if there's no meta tag on the page? Just `{}`. And we don't actually need to worry about whether or not there's already an X-CSRF-Token in the custom headers, since it'll just overwrite the automatic one in the spread.

I think this can be

```js
export default function csrfHeader(): { 'X-CSRF-Token': string }
  const token = getToken()
  if (token == null) return {}
  return { 'X-CSRF-Token': token }
}
```

and then at the call site we can just say

```js
headers: new Headers({
  Accept: 'application/json',
  'Content-Type': 'application/json',
  ...csrfHeader(),
  ...headers,
}),
```

I think that's a little clearer... What do you think?